File: API/DTOs/OrderDto.cs
using Core.Entities.OrderAggregate;
using System;
using System.Collections.Generic;

namespace API.DTOs
{
    public class OrderDto
    {
        public int Id { get; set; }
        public DateTime OrderDate { get; set; }
        public required string BuyerEmail { get; set; }
        public required ShippingAddressDto ShippingAddress { get; set; }
        public required string DeliveryMethod { get; set; }
        public decimal ShippingPrice { get; set; }
        public required PaymentSummaryDto PaymentSummary { get; set; }
        public required List<OrderItemDto> OrderItems { get; set; }
        public decimal Subtotal { get; set; }
        public required string PaymentIntentId { get; set; }
        public required string Status { get; set; }
        public decimal Total { get; set; }
    }
}

File: java-api/api/src/main/java/com/aayuskinet/api/dtos/OrderDto.java
package com.aayuskinet.api.dtos;

import com.aayuskinet.core.entities.orderaggregate.OrderStatus;
import lombok.Data;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.List;

@Data
public class OrderDto {
    private int id;
    private OffsetDateTime orderDate;
    private String buyerEmail;
    private ShippingAddressDto shippingAddress;
    private String deliveryMethod;
    private BigDecimal shippingPrice;
    private PaymentSummaryDto paymentSummary;
    private List<OrderItemDto> orderItems;
    private BigDecimal subtotal;
    private String paymentIntentId;
    private String status; // Using String for status as in C# DTO
    private BigDecimal total;
}
